Album/ViewControllers/ProfileControllers/ActivitiesViewController.m

```diff
@@ -6,41 +6,77 @@
 //
 
 #import "ActivitiesViewController.h"
-
-@interface ActivitiesViewController ()<UITableViewDataSource, UITableViewDelegate>
+#import "FriendProfileViewController.h"
+#import "ActivityCell.h"
+#import "Parse/Parse.h"
+#import "AlbumConstants.h"
+@interface ActivitiesViewController ()<UITableViewDataSource, UITableViewDelegate, ActivityCellDelegate>
 @property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSMutableArray *requests;
 
 @end
 
 @implementation ActivitiesViewController
-
+- (void) viewWillAppear:(BOOL)animated {
+	[super viewWillAppear: animated];
+	[self fetchActivities];
+	[self.tableView reloadData];
+}
 - (void)viewDidLoad {
-    [super viewDidLoad];
-    // Assign tableview data source and delegate
-    self.tableView.delegate = self;
-    self.tableView.dataSource = self;
+	[super viewDidLoad];
+	// Assign tableview data source and delegate
+	self.tableView.delegate = self;
+	self.tableView.dataSource = self;
+	self.requests = [[NSMutableArray alloc] init];
+	self.tableView.rowHeight = UITableViewAutomaticDimension;
 }
 - (IBAction)backButton:(id)sender {
-    [self dismissViewControllerAnimated:YES completion:nil];
+	[self dismissViewControllerAnimated:YES completion:nil];
 }
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
-    return 0;
+
+- (void) fetchActivities {
+	[self fetchRequests];
 }
 
-// Row display. Implementers should *always* try to reuse cells by setting each cell's reuseIdentifier and querying for available reusable cells with dequeueReusableCellWithIdentifier:
-// Cell gets various attributes set automatically based on table (separators) and data source (accessory views, editing controls)
+- (void) fetchRequests {
+	// Query to find pending friend requests for this user
+	self.requests = [[NSMutableArray alloc] init];
+	PFQuery *query = [PFQuery queryWithClassName:classNameFriendship];
+	PFUser *currentUser = [PFUser currentUser];
+	[query whereKey:@"recipientId" equalTo:currentUser.objectId];
+	[query whereKey:@"hasFriended" equalTo:@(PENDING)];
+	[query orderByDescending:@"updatedAt"];
+	[query findObjectsInBackgroundWithBlock:^(NSArray *friendships, NSError *error) {
+	         if (friendships != nil) {
+			 for(PFObject *friendship in friendships) {
+				 PFQuery *query = [PFUser query];
```

Comment:
you create d a new query, but we are never calling the query. 
How come query contains the object `user` with [query findObjects][0]?